setup.py

```diff
@@ -1,74 +0,0 @@
-#!/usr/bin/env python
-#
-# Copyright 2017-2021 European Centre for Medium-Range Weather Forecasts (ECMWF).
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#   http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-import os
-import re
-
-import setuptools  # type: ignore
-
-
-def read(path: str) -> str:
-    file_path = os.path.join(os.path.dirname(__file__), *path.split("/"))
-    return open(file_path).read()
-
-
-# single-sourcing the package version using method 1 of:
-#   https://packaging.python.org/guides/single-sourcing-package-version/
-def parse_version_from(path: str) -> str:
-    version_file = read(path)
-    version_match = re.search(r'^__version__ = "(.*)"', version_file, re.M)
-    if version_match is None or len(version_match.groups()) > 1:
-        raise ValueError("couldn't parse version")
-    return version_match.group(1)
-
-
-setuptools.setup(
-    name="cfgrib",
-    version=parse_version_from("cfgrib/__init__.py"),
-    description="Python interface to map GRIB files to the NetCDF Common Data Model "
-    "following the CF Convention using ecCodes.",
-    long_description=read("README.rst") + read("CHANGELOG.rst"),
```

Comment:
README.rst made it over, CHANGELOG.rst did not.  Setuptools seems to have [a method](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata) for doing this, but [hatch explicitly disallows that approach](https://github.com/pypa/hatch/blob/c7575ad551f380bd2a1078bb5ba2855bf961a659/backend/src/hatchling/metadata/core.py#L560-L563)